client/mysqldump.c

```diff
@@ -4029,6 +4038,78 @@ static void vers_append_system_time(DYNAMIC_STRING* query_string)
   dynstr_append_checked(query_string, "'");
 }
 
+static int add_header_to_output_file(const char *table, const char *db, char *filename, 
+                                      char *enclosed, char *opt_enclosed,
+                                      char *fields_terminated, char *lines_terminated) {
+  char query_string_for_columns[NAME_LEN*2+200];
+  char filename_headers[FN_REFLEN];
+  char enclosing_str[NAME_LEN];
+  FILE *tmp_header_file, *outfile;
+  my_bool first_col;
+  unsigned char buff[8192];
+  size_t n, m;
+  MYSQL_ROW row;
+
+  MYSQL_RES *column_res;
+
+  sprintf(query_string_for_columns, "SELECT `COLUMN_NAME` FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA`='%s' AND `TABLE_NAME`='%s'", db, table);
+  if (mysql_query_with_error_report(mysql, &column_res, query_string_for_columns)) 
+  {
```

Comment:
The typical coding style in MariaDB does _not_ include braces (`{ … }`) for single-statement `if` blocks.

Numerous other examples below…